File: An2/Sem1/MAP/MAP-C#/Partial/Partial/Repository/Repository.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Runtime.Serialization.Formatters.Binary;
using System.Text;
using System.Threading.Tasks;

using Partial.Model;

namespace Partial.Repository
{
    class Repository<T> : RepoInterface<T> where T : Leguma
    {
        private LinkedList<T> elements;

        public LinkedList<T> getElements()
        {
            return this.elements;
        }

        public void writeToFile(LinkedList<T> elementsToWrite)
        {
            try
            {
                StreamWriter writer = new StreamWriter("output.txt");
                foreach (T elem in elementsToWrite)
                {
                    writer.WriteLine(elem.toString());
                }
                writer.Close();
            }
            catch (IOException e)
            {
                Console.Out.WriteLine(e.Message);
            }
        }

        public void replaceContent(LinkedList<T> content)
        {
            elements = content;
        }

        public void serializeDataToFile(string fileName)
        {
            Stream fileStream = File.Create(fileName);
            BinaryFormatter serializer = new BinaryFormatter();
            serializer.Serialize(fileStream, this.elements);
            fileStream.Close();
        }

    }
}

File: An2/Sem1/MAP/MAP-Java/Exam/src/Repository/Repository.java
package Repository;

import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;

/**
 * Created by mihai on 2/3/14.
 */
public class Repository<T> implements RepoInterface<T> {

    private ArrayList<T> elements = new ArrayList<T>();

    @Override
    public ArrayList<T> getElements() {
        return elements;
    }

    @Override
    public void writeToFile(ArrayList<T> elementsToWrite) {
        try{
            FileWriter out = new FileWriter("output.txt");
            for (Object elem : elementsToWrite){
                out.write(elem.toString());
                out.write("\n");
            }
            out.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void replaceContent(ArrayList<T> content) {
        elements = content;
    }

    @Override
    public void addElement(T element) {
        this.elements.add(element);
    }


}
